src/ncp/ncp_base.cpp

```diff
@@ -1000,22 +1000,30 @@ otError NcpBase::HandleCommand(uint8_t aHeader, unsigned int aCommand, const uin
         SPINEL_HEADER_GET_IID(aHeader) == 0,
         error = SendLastStatus(aHeader, SPINEL_STATUS_INVALID_INTERFACE)
     );
-
-    for (i = 0; i < sizeof(mCommandHandlerTable) / sizeof(mCommandHandlerTable[0]); i++)
+#if OPENTHREAD_ENABLE_SPINEL_VENDOR_SUPPORT
+    if (aCommand >= SPINEL_CMD_VENDOR__BEGIN && aCommand <= SPINEL_CMD_VENDOR__END)
```

Comment:
nit: the `SPINEL_<SOMETHING>__END` is not inclusive, so the second test in the `if` would be better as  `aCommand < SPINEL_CMD_VENDOR__END`  instead of `aCommand <= SPINEL_CMD_VENDOR__END`.